### PR TITLE
[FIX] Fix from_chars for float 

### PR DESCRIPTION
**Pull request by smehringer**
Treat the second parameter as the end of the sequence to parse.

### CODE REVIEW

File: include/seqan3/std/charconv_detail.hpp [hunk 1]
```diff
@@ -724,11 +724,11 @@ inline std::from_chars_result from_chars_floating_point(char const * first,
     if (*first == '+') // + is permitted in function strto[d/f/ld] but not in from_chars
         return {last, std::errc::invalid_argument};
 
+    using ptrdiff_type = decltype(last - first);
```
Comment (h-2) [inline include/seqan3/std/charconv_detail.hpp]: why is this necessary?

File: include/seqan3/std/charconv_detail.hpp [hunk 2]
```diff
@@ -758,40 +758,46 @@ inline std::from_chars_result from_chars_floating_point(char const * first,
...
 
+    // In contrast to std::from_chars, std::strto[f/d/ld] does not treat the second
+    // parameter (str_end) as "end of the sequence to parse" but merely as an out
+    // parameter to indicate where the parsing ended. Therefore, a buffer is
+    // needed to represent the truncated sequence and ensure correct from_chars
+    // functionality.
+    ptrdiff_type str_length{std::min(ptrdiff_type{buffer_size - offset}, last - first)};
+
+    for (ptrdiff_type i = 0; i < str_length; ++i)
+        buffer[i+offset] = first[i];
```
Comment (h-2) [inline include/seqan3/std/charconv_detail.hpp]: use std::copy?

File: include/seqan3/std/charconv_detail.hpp [hunk 3]
```diff
@@ -758,40 +758,46 @@ inline std::from_chars_result from_chars_floating_point(char const * first,
...
 
     if (errno == ERANGE)
     {
         return {last, std::errc::result_out_of_range};
     }
-    else if (tmp == 0 && end == first)
+    else if (tmp == 0 && end == &buffer[0])
     {
         return {last, std::errc::invalid_argument};
     }
```
Comment (h-2) [inline include/seqan3/std/charconv_detail.hpp]: As discussed I would recommend only copying to the buffer when the input is not alread 0-terminated.

File: include/seqan3/std/charconv_detail.hpp (hunk 1 above)
Comment (smehringer) [reply include/seqan3/std/charconv_detail.hpp]: I changed it back

File: include/seqan3/std/charconv_detail.hpp [hunk 4]
```diff
@@ -87,6 +87,7 @@
 #include <cstring>
 #include <cmath>
 #include <cerrno>
+#include <algorithm>
```
Comment (h-2) [inline include/seqan3/std/charconv_detail.hpp]: order

File: test/unit/std/charconv_test.cpp [hunk 5]
```diff
@@ -270,16 +280,18 @@ TYPED_TEST(from_char_real_test, real_numbers)
         TypeParam val{};
         std::string str = "1234";
         auto res = std::from_chars(&str[0], &str[0] + str.size(), val);
-        EXPECT_EQ(val, TypeParam{1234});
+        EXPECT_TRUE(double_equals(val, TypeParam{1234}));
```
[the remark this answers is not in the document]
Comment (smehringer) [reply test/unit/std/charconv_test.cpp]: cool thanks! I didn't know they have this

[the remark this answers is not in the document]
Comment (smehringer) [reply test/unit/std/charconv_test.cpp]: .... My tests fail with `EXPECT_DOUBLE_EQ` from google test since the epsilon is too small :scream: 
```
  val // from_chars output
    Which is: -0.30000001192092896
  TypeParam{-0.3} // direct initialization
    Which is: -0.29999999999999999
```

... @h-2 @joergi-w Do you think this is too severe? ... But I think if anything this is the fault of `std::strtof`

[the remark this answers is not in the document]
Comment (joergi-w) [reply test/unit/std/charconv_test.cpp]: Yes, that's exactly what I would expect :)

[the remark this answers is not in the document]
Comment (smehringer) [reply test/unit/std/charconv_test.cpp]: alright float then
